Sources/Commands/Workspace.swift

```diff
@@ -301,6 +304,10 @@ public class Workspace {
         try localFileSystem.createDirectory(repositoriesPath, recursive: true)
         try localFileSystem.createDirectory(checkoutsPath, recursive: true)
 
+        // Create a file lock and immediately try to aquire the lock before restoring the previous state.
+        fileLock = FileLock(name: "swiftpm-workspace-lock", cachePath: self.dataPath)
+        try fileLock.lock()
```

Comment:
I would probably implement some ramp approach, where we wait 1s with no notification, then tell you we are waiting, then wait 10s, then wait 60s in a loop and tell you we are still waiting each time.

Seems ad hoc, but I don't have a better idea.